skbio/sequence/_grammared_sequence.py

```diff
@@ -22,11 +22,79 @@
 from ._sequence import Sequence
 
 
-class IUPACSequence(with_metaclass(ABCMeta, Sequence)):
-    """Store biological sequence data conforming to the IUPAC character set.
+class GrammaredSequenceException(TypeError):
```

Comment:
Sorry I wasn't clearer before, I'd just remove this custom exception type and raise `TypeError` directly in the validations.